### PR TITLE
Remove loki dependency, copy files from logproto

### LINKED ISSUE #16821: Release blocked on dependencies
### Component(s)

_No response_

### Describe the issue you're reporting

The contrib release is currently blocked. The following steps will be taken to address the blocking issue:

1. revert the loki dependency on promtail
2. downgrade prometheus to v0.39.1

### PR DESCRIPTION
**Pull request by bogdandrutu**
Files copied from `github.com/grafana/loki/pkg/logproto` to remove unnecessary dependencies. In `logproto.pb.go` I had to remove few types `Query[Request|Response]` and `SampleQuery[Request|Response]` and the gRPC service that uses them, because they depend on another loki package `stats`.

Fixes #16821

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
This seems ok, so long at the code owners for the loki exporter are ok with this approach. Is the gopsutil downgrade needed in this pr?

**Comment by Aneurysm9** [comment]
> This seems ok, so long at the code owners for the loki exporter are ok with this approach. Is the gopsutil downgrade needed in this pr?

That tag was removed from the gopsutil repo so either the downgrade is needed or an `exclude` directive added, which will have the same effect as there is no newer version available.

**Comment by codeboten** [comment]
I pulled the downgrade into a separate pr: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16824

**Review by jpkrohling — APPROVED** [review]
This is very similar to what I have in a local WIP branch, with the addition that I have a readme in the proto directory outlining what changes were made, so that whenever it gets updated, we know how to re-do these changes. Given that this seems to be blocking the release, I'm giving my +1.

I was currently trying to backport the changes to the proto files and re-generating the Go files based on that so that we wouldn't make manual changes to the generated files, but that can wait.

**Review by codeboten — APPROVED** [review]

File: receiver/dockerstatsreceiver/factory.go
Comment (codeboten) [inline receiver/dockerstatsreceiver/factory.go:79]: This change is unrelated

Comment (bogdandrutu) [reply receiver/dockerstatsreceiver/factory.go:79]: HATEEEE GOLAND renme.

**Comment by mar4uk** [comment]
@bogdandrutu could you please clarify why this change was necessary?
I see from the liked issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16821 that release was blocked because of 
- revert the loki dependency on promtail
- downgrade prometheus to v0.39.1

Promtail receiver was excluded from the release and that unblocked the release.
The Prometheus version is currently v0.40.1 in the collector-contrib. So from my understanding downgrading to v0.39.1 was not done and isn't needed anymore.

Can we revert this PR? and use logproto from loki directly instead of copying files? What cons do you see in this approach.
Does the [replace statements](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16722) block importing `github.com/grafana/loki/pkg/logproto` right now? Will it be ok to import `github.com/grafana/loki/pkg/logproto` after https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16722 is solved?

**Pull request merged**
